staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3675,6 +3690,74 @@ func buildListener(port v1.ServicePort, annotations map[string]string, sslPorts
 	return listener, nil
 }
 
+func (c *Cloud) parseNLBHealthCheckAnnotations(portMapping *nlbPortMapping, annotations map[string]string) error {
+	getStringValue := func(annotation string, defaultVal string) string {
+		if v, ok := annotations[annotation]; ok {
+			return v
+		}
+		return defaultVal
+	}
+
+	getValueOrDefault := func(annotation string, defaultVal int64) (int64, error) {
+		i64 := defaultVal
+		if s, ok := annotations[annotation]; ok {
+			var err error
+			i64, err = strconv.ParseInt(s, 10, 0)
+			if err != nil {
+				return i64, fmt.Errorf("failed to parse health check annotation value: %v", err)
+			}
+		}
+		return i64, nil
+	}
+
+	portMapping.HealthCheckProtocol = strings.ToUpper(getStringValue(ServiceAnnotationLoadBalancerHealthCheckProtocol, defaultHealthCheckProtocol))
+	switch portMapping.HealthCheckProtocol {
+	case elbv2.ProtocolEnumHttp, elbv2.ProtocolEnumHttps:
+		portMapping.HealthCheckPath = getStringValue(ServiceAnnotationLoadBalancerHealthCheckPath, defaultHealthCheckPath)
+	case elbv2.ProtocolEnumTcp:
+	default:
+		return fmt.Errorf("Invalid Health check protocol %v", portMapping.HealthCheckProtocol)
+	}
+
+	portStr := getStringValue(ServiceAnnotationLoadBalancerHealthCheckPort, defaultHealthCheckPort)
+
+	if portStr != "traffic-port" {
```

Comment:
if it's set to 'traffic-port' explicitly, the healthcheck is still using the nodePort, is it expected? (the effect should be same, but users can see it's a port number instead of 'traffic-port' in aws console

Reply:
HealthCheckPort is stored as int64 in nlbPortMapping, so the conversion. I can change it to traffic-port if it is more helpful to the end users.